srcopsmetrics/github_knowledge.py

```diff
@@ -143,21 +168,12 @@ def get_referenced_issues(pull_request: PullRequest) -> List[int]:
         """
         issues_referenced = []
         for comment in pull_request.get_issue_comments():
-            message = comment.body.split(" ")
-            for idx, word in enumerate(message):
-                if word.replace(":", "") in ISSUE_KEYWORDS:
-                    try:
-                        _LOGGER.info("      ...found keyword referencing issue")
-                        referenced_issue_number = message[idx + 1]
-                        assert (referenced_issue_number).startswith("https")
-                        # last element of url is always the issue number
-                        issues_referenced.append(referenced_issue_number.split("/")[-1])
-                        _LOGGER.info("      ...referenced issue number present")
-                        # we assure that this was really referenced issue
-                        # and not just a keyword without number
-                    except (IndexError, AssertionError) as e:
-                        _LOGGER.info("      ...referenced issue number absent")
-                        _LOGGER.debug(str(e))
+            for id in self.search_for_references(comment.body):
+                issues_referenced.append(id)
+
+        for id in GitHubKnowledge.search_for_references(pull_request.body):
```

Comment:
also here should with be `self.search_for_references` sorry :)

Reply:
hmm I still cannot make a sense out of it...

we previously had two static functions -> ```get_referenced_issues``` and ```search_for_references```, where the first function calls the second. My reasoning behind the use of ```@staticmethod``` was, that within these functions, we don't use the object itself ( ```self``` ). Therefore, to avoid bounding functions```get_referenced_issues``` and ```search_for_dependencies``` to each instantiation of the class ```GitHubKnowledge()```, these functions were made static.

Maybe @fridex @goern @CermakM can join the conversation and write their opinion about it?

Reply:
IMHO, if `get_referenced_issues` is in that class and uses another method of the same class it should not be static, otherwise, it should not be in that class at all. But I'm not the expert here :) Open for suggestions/opinions :)

Reply:
Well first of all, `@staticmethod`s are used primarily for two reasons -- to fool the linter not to complain that there is unused self and to denote a function which can or should be called on a class without creating an instance of that class. In your particular case, passing the `self` and calling the static functions using `self` makes total sense, as you no longer have to fool the interpreter and your intention was not to emphasize that the function should be called on the class instead of on the instance.
However, what you apparently have here is a class simulating a `namespace` concept (given that basically all of the functions present there are static ones). For these reasons, python has modules. Is there any particular reason why you introduced that class and why not to have the functions encapsulated only by the module scope?

Reply:
> Is there any particular reason why you introduced that class and why not to have the functions encapsulated only by the module scope?

Yes, in near future the class would receive a minor refactor update, in which it would be fixed to be fully used as a proper class.